src/BaseHostPlugin.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Allow to modify the the schema and the host of an existing request.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class BaseHostPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $baseUri;
+
+    /**
+     * @var bool
+     */
+    private $replace;
+
+    /**
+     * Available options for $config are:
+     *  - replace: bool True will replace all hosts, false will only add host when none is specified.
+     *
+     * @param UriInterface $baseUri
+     * @param array        $config
```

Comment:
Yeah, I saw that but I choose not to include it in this PR because I wanted to be consistent with other plugins and then send a PR to fix them all. 

I'll make sure this plugin is doing it the correct way now already then. =)